.github/workflows/nightly.yml

```diff
@@ -1,121 +1,156 @@
+          fi
+
+        fi
+
+        NEW_BASE_COMMIT=$(git log $BASE_BRANCH -n 1 --format=%H)
+
+        echo "Choosing base branch $BASE_BRANCH because it's newer"
+        echo "$BASE_BRANCH commit ID is $NEW_BASE_COMMIT"
+
+        if [[ "$OLD_BASE_COMMIT" == "$NEW_BASE_COMMIT" && "$GH_EVENT_NAME" == "pull_request" ]]; then
+          echo "Nightly already points to same base; leave unchanged";
+          echo "base-commit=skip" >> "$GITHUB_OUTPUT"
+          exit 0
+        elif [[ "$GH_EVENT_NAME" == "push" ]]; then
+          LATEST_TAG=$(git describe --tags --abbrev=0)
+
+          echo Looking for non-chore commits since last tag
+          echo
+          git --no-pager log --pretty=format:"%s" $LATEST_TAG..HEAD | grep -v "^chore"
+          if [ $? -eq 0 ]; then
+            echo
+            echo "Found non-chore commits; expecting release-plz to create a new release branch."
+            echo "base-commit=skip" >> "$GITHUB_OUTPUT"
+            exit 0
+          else
+            echo "Push to main, but there are only chore commits since last tag."
+            echo "Not expecting a new release branch"
+            echo;
+          fi
+        fi
+
+        echo "Updating nightly base";
+        echo "base-commit=$NEW_BASE_COMMIT" >> "$GITHUB_OUTPUT"
 
-    - name: Bump crate versions
-      run: |
-        sed -i "s/^version = \"[^\"]*\"$/version = \"$VERSION\"/;" sdk/Cargo.toml
-      env:
-        VERSION: ${{ steps.set-nightly-version.outputs.version }}
-    
     - name: Install Rust toolchain
+      if: ${{ steps.choose-base.outputs.base-commit != 'skip' }}
       uses: dtolnay/rust-toolchain@stable
       with:
         components: cargo
 
-    - name: Create or update Cargo.lock
+    - name: Add -nightly+(date)-(commit ID) prefix to crate versions
+      id: set-nightly-version
+      if: ${{ steps.choose-base.outputs.base-commit != 'skip' }}
       run: |
+        echo Force updating nightly branch to point to base branch
+        git branch -f nightly
+
+        export NIGHTLY_SUFFIX=$(echo -nightly+`date +%F`-`git rev-parse --short HEAD`)
+        echo Will add nightly suffix $NIGHTLY_SUFFIX
+
+        sed -i "s/^version = \"\\(.*\\)\"/version = \"\\1$NIGHTLY_SUFFIX\"/" sdk/Cargo.toml
+
         cargo update -w
         git add -f Cargo.lock
 
-    - name: Report differences for "prepare (release)" commit
-      run: git diff
+        echo
+        echo Proposed changes:
+        git status
+        git diff
 
     - name: Commit Cargo.toml and Cargo.lock
       uses: stefanzweifel/git-auto-commit-action@v5
       id: commit
+      if: ${{ steps.choose-base.outputs.base-commit != 'skip' }}
       with:
         branch: nightly
         push_options: '--force'
-        commit_message: Prepare ${{ steps.set-nightly-version.outputs.version }} release
+        commit_message: Prepare nightly release
         commit_user_name: Adobe CAI Team
         commit_user_email: noreply@adobe.com
         create_branch: true
-
-  tests:
```

Comment:
Removed this custom tests job in favor of running the full CI job against the nightly branch.